src/iso15118/io/socket_helper.cpp

```diff
@@ -21,6 +22,7 @@ bool get_first_sockaddr_in6_for_interface(const std::string& interface_name, soc
     auto current_if = if_list_head;
 
     bool found_interface = false;
+    char buffer[INET6_ADDRSTRLEN];
```

Comment:
buffer is only used in the `if (interface_name == "auto")`, hence move it there

Reply:
`buffer` is not longer needed